providers/signalk-ws.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 Ilker Temir <ilker@ilkertemir.com> based on code
+ * by Teppo Kurki <teppo.kurki@iki.fi>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var Transform = require('stream').Transform;
+
+var SignalK = require('signalk-client');
+
+var debug = require('debug')('signalk-server:providers:signalk-ws');
+
+var WebSocket = require('ws');
+var _object = require('lodash/object');
+
+function SignalKWs(options) {
+  Transform.call(this, {
+    objectMode: true
+  });
+  this.selfHost = options.app.config.getExternalHostname() + ".";
+  this.selfPort = options.app.config.getExternalPort();
+  this.remoteServers = {};
+  this.remoteServers[this.selfHost + ":" + this.selfPort] = {};
+  this.signalkClient = new SignalK.Client();
+  var url = "ws://" + options.host + ":" + options.port + options.path;
```

Comment:
We have been discussing this on Slack but I am not sure if I am convinced about this. There is an underlying assumption here that the path is always /signalk/v1/stream and I do understand you want a static path because you want mDNS discovery, which only provides host+port.

However, this wasn't clear to me when I was writing my signalk server and I initially used a different path. I switched it to the standard later on but I can still see cases where there will be multiple paths.

In the case of mDNS, I think there are several options for not requiring a static path. Some options, not necessarily exact:

- Use a GET to obtain WS path, i.e. http://host:port/wsPath gives you that. This is easier and clearer to implement. There can be a default as well.
- Insert the WS path into the mDNS TXT record. 

I personally don't think that assuming the path will always be the same is not right.